build.gradle.kts

```diff
@@ -0,0 +1,221 @@
+    doLast {
+        printReportSafe(findbugsMain.reports.xml.destination)
+    }
+}
+val findbugsTest: FindBugs by tasks
+val findbugsTestReport = task("findBugsTestReport") {
+    doLast {
+        printReportSafe(findbugsTest.reports.xml.destination)
+    }
+}
+findbugsMain.finalizedBy(findbugsMainReport)
+findbugsTest.finalizedBy(findbugsTestReport)
+
+tasks.withType<JacocoReport> {
+    reports {
+        xml.isEnabled = true
+        html.isEnabled = true
+    }
+}
+
+tasks.withType<Test> {
+    testLogging {
+        if (project.hasProperty("logTests") || project.hasProperty("jenkinsBuild")) {
+            events("started", "passed", "skipped", "failed")
+        } else {
+            events("failed")
+        }
+        exceptionFormat = TestExceptionFormat.FULL
+    }
+    /*
+     * Allows you to run the UI tests in headless mode by calling gradle with the -Pheadless argument
+     */
+    if (project.hasProperty("jenkinsBuild") || project.hasProperty("headless")) {
+        println("Running UI Tests Headless")
+
+        jvmArgs = listOf(
+            "-Djava.awt.headless=true",
+            "-Dtestfx.robot=glass",
+            "-Dtestfx.headless=true",
+            "-Dprism.order=sw",
+            "-Dprism.text=t2k"
+        )
+        useJUnit {
+            this as JUnitOptions
+            excludeCategories("edu.wpi.first.shuffleboard.NonHeadlessTests")
+        }
+    }
+
+}
+
+application {
+    mainClassName = "edu.wpi.first.shuffleboard.Shuffleboard"
+}
+
+val sourceJar = task<Jar>("sourceJar") {
+    description = "Creates a JAR that contains the source code."
+    from(java.sourceSets["main"].allSource)
+    classifier = "sources"
+}
+
+tasks.withType<Jar> {
+    if (WPILibVersion.version != "") {
+        version = WPILibVersion.version
+    }
+}
+
+publishing {
+    publications {
+        create<MavenPublication>("shadow") {
+            groupId = "edu.wpi.first.shuffleboard"
+            artifactId = "Shuffleboard"
+            if (WPILibVersion.version != "") {
+                version = WPILibVersion.version
+            }
+            from(components["java"])
+            artifact(sourceJar)
+        }
+    }
+}
+
+// Ensure that the WPILibVersioningPlugin is setup by setting the release type, if releaseType wasn't
+// already specified on the command line
+if (!hasProperty("releaseType")) {
+    WPILibVersion {
+        releaseType = ReleaseType.DEV
+    }
+}
+
+task<Wrapper>("wrapper") {
+    gradleVersion = "4.0.2"
+}
+
+/**
+ * Retrieves the [java][org.gradle.api.plugins.JavaPluginConvention] project convention.
+ */
+val Project.`java`: org.gradle.api.plugins.JavaPluginConvention get() =
+convention.getPluginByName<org.gradle.api.plugins.JavaPluginConvention>("java")
+
+
+/**
+ * Configures the [checkstyle][org.gradle.api.plugins.quality.CheckstyleExtension] project extension.
+ */
+fun Project.`checkstyle`(configure: org.gradle.api.plugins.quality.CheckstyleExtension.() -> Unit): Unit =
```

Comment:
This is what you get printed when you run the `gskGenAccessors` task. I just copied and pasted them. 

There's a different mode you can use that doesn't require these declarations in this file but I've never tried it. I could try it and see if it works.